websites/G/galaxy/presence.ts

```diff
@@ -0,0 +1,462 @@
+/* eslint-disable no-one-time-vars/no-one-time-vars */
+const presence = new Presence({
+		clientId: "1122404044829380739",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/KQvYGoh.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ host, pathname } = document.location,
+		path = pathname.slice(1).split("/"),
+		showTimestamp = await presence.getSetting<boolean>("timestamp"),
+		showButtons = await presence.getSetting<boolean>("buttons"),
+		makeURL = (p: string) => `https://${host}${p}`;
+
+	switch (path[0]) {
+		case "": {
+			presenceData.details = "viewing the front page";
+			presenceData.buttons = [
+				{
+					label: "view page",
+					url: makeURL(""),
+				},
+			];
+			break;
+		}
+		// Game pages
+		case "play": {
+			const gameId = path[1];
+			if (gameId) {
+				const gameName =
+						document.querySelector<HTMLHeadingElement>(
+							"#game-title"
+						)?.textContent,
+					currentLevel = document.querySelector<HTMLSpanElement>(
+						"#progress > .level > span"
+					)?.textContent,
+					// "N / D"
+					xpRatio = document.querySelector<HTMLSpanElement>(
+						"#progress > .xp > span"
+					)?.textContent;
+
+				presenceData.details = gameName
+					? `playing ${gameName}`
+					: "playing a game";
+				if (currentLevel && xpRatio) {
+					const [currentXP, neededXP] = xpRatio.split("/").map(n => n.trim());
+
+					presenceData.state = `level ${currentLevel} | ${currentXP}/${neededXP} xp`;
+				}
+				presenceData.buttons = [
+					{
+						label: "play game",
+						url: makeURL(`/play/${gameId}`),
```

Comment:
why not use [href ](https://developer.mozilla.org/en-US/docs/Web/API/Location/href)instead?

Reply:
:+1: